packages/aws-cdk-lib/aws-ec2/lib/nat.ts

```diff
@@ -436,10 +444,33 @@ class PrefSet<A> {
  * The instance uses Amazon Linux 2023 as the operating system.
  */
 export class NatInstanceProviderV2 extends NatProvider implements IConnectable {
+  /**
+   * Amazon Linux 2023 NAT instance user data commands
+   * Enable iptables on the instance, enable persistent IP forwarding, configure NAT on instance
+   * @see https://docs.aws.amazon.com/vpc/latest/userguide/VPC_NAT_Instance.html#create-nat-ami
+   */
+  public static readonly DEFAULT_USER_DATA_COMMANDS = [
+    'yum install iptables-services -y',
+    'systemctl enable iptables',
+    'systemctl start iptables',
+    'echo "net.ipv4.ip_forward=1" > /etc/sysctl.d/custom-ip-forwarding.conf',
+    'sudo sysctl -p /etc/sysctl.d/custom-ip-forwarding.conf',
+    "sudo /sbin/iptables -t nat -A POSTROUTING -o $(route | awk '/^default/{print $NF}') -j MASQUERADE",
```

Comment:
I have tested to manually change this line in `node_modules` and it worked! Before that I was getting timeouts for all resources which tried to route traffic through NatInstance V2.

Thanks @nmussy and I hope this will gets merged soon - the deprecation notices are really annoing especially when the solution they provide does not work at all.